libnetwork/network.go

```diff
@@ -1987,7 +1987,21 @@ func (n *network) ResolveName(req string, ipType int) ([]net.IP, bool) {
 
 	req = strings.TrimSuffix(req, ".")
 	req = strings.ToLower(req)
-	ipSet, ok := sr.svcMap.Get(req)
+
+	// Support wildcard matching, naïve implementation uses a loop
+	selectedKey := req
+	ok = false
+	var ipSet []interface{}
+	for _, key := range sr.svcMap.Keys() {
+		if key == selectedKey ||
+			(strings.HasPrefix(key, "*.") &&
+				strings.HasSuffix(req, strings.TrimPrefix(key, "*"))) {
+			selectedKey = key
+			ok = true
+			ipSet, _ = sr.svcMap.Get(selectedKey)
```

Comment:
```suggestion
			ipSet, found = sr.svcMap.Get(selectedKey)
			if !found  {
				// The key got deleted
				continue
			}
```

Just a little sanity check (see [here](https://github.com/moby/moby/blob/34c296454b2f7a966ce03114765d7251a3b61750/libnetwork/network.go#L1498-L1501))

Reply:
Good spot, thanks! 

In theory this is unnecessary due to the controller lock `c.Lock()` above. Should we should also log a warn if this happens maybe?

Reply:
Fixed in https://github.com/moby/moby/pull/43444/commits/12171050bfeb6346f1b171b870ceaaee1ff358ab

Reply:
Cool!